src/libasr/codegen/asr_to_fortran.cpp

```diff
@@ -870,7 +870,20 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
 
     // void visit_ArraySection(const ASR::ArraySection_t &x) {}
 
-    // void visit_ArraySize(const ASR::ArraySize_t &x) {}
+    void visit_ArraySize(const ASR::ArraySize_t &x) {
+        visit_expr(*x.m_v);
+        std::string r = "";
+        if (x.m_dim == nullptr) {
+            // TODO: return the product of all dimensions:
+            r = "0";
+        } else {
+            if( x.m_dim ) {
+                visit_expr(*x.m_dim);
+                r += s + "-1";
+            }
+        }
```

Comment:
I would write this as:
```suggestion
        std::string r = "size(" + s;
        if (x.m_dim) {
            r += ", ";
            for (int i=0; i < x.n_dim; i++) {
                visit_expr(*x.m_dim);
                r += s;
                if (i < x.n_dim-1) r += ", ";
            }
        }
        r += ")";
```